File: plugin/src/shared/sc/plugin2019/FieldState.java
package sc.plugin2019;

import sc.shared.PlayerColor;

import java.util.Objects;

public enum FieldState {
  RED,
  BLUE,
  OBSTRUCTED,
  EMPTY;
  public static FieldState from(PlayerColor color) {
    switch(color) {
      case RED: return RED;
      case BLUE: return BLUE;
      default: throw new IllegalArgumentException("PlayerColor can only be RED or BLUE");
    }
  }
}

File: plugin/src/shared/sc/plugin2019/FieldState.kt
package sc.plugin2019

import sc.shared.PlayerColor

enum class FieldState {
    RED,
    BLUE,
    OBSTRUCTED,
    EMPTY;

    override fun toString() = when(this) {
        OBSTRUCTED -> "Obstructed"
        RED -> "Red"
        BLUE -> "Blue"
        else -> "empty"
    }

    fun asLetter() = when(this) {
        OBSTRUCTED -> "O"
        RED -> "R"
        BLUE -> "B"
        else -> " "
    }

    companion object {
        fun from(color: PlayerColor): FieldState {
            return when(color) {
                PlayerColor.RED -> RED
                PlayerColor.BLUE -> BLUE
            }
        }
    }
}
